media/iso_iec_base_media.ksy

```diff
@@ -153,6 +178,261 @@ types:
+            type: str
+            terminator: 0
+            if: _parent.type == 'urn '
+          - id: location
+            type: str
+            doc: |
+              This field is probably incorrect sometimes (GDCL muxer seems to have a bug here)
+            terminator: 0
+            eos-error: false
+  elst_body:
+    seq:
+      - id: version
+        type: u4
+      - id: entry_count
+        type: u4
+      - id: entries
+        type: entry
+        repeat: expr
+        repeat-expr: entry_count
+    types:
+      entry:
+        seq:
+          - id: segment_duration
+            type:
+              switch-on: _parent.version
+              cases:
+                1: s8
+                _: s4
+          - id: media_time
+            type:
+              switch-on: _parent.version
+              cases:
+                1: s8
+                _: s4
+          - id: media_rate_integer
+            type: u2
+          - id: media_rate_fraction
+            type: u2
+  hdlr_body:
+    seq:
+      - id: predefined
+        type: u4
+      - id: reserved0
+        type: u4
+      - id: reserved1
+        type: u4
+      - id: reserved2
+        type: u4
+  mdhd_body:
+    seq:
+      - id: version
+        type: u4
+      - id: creation_time
+        type:
+          switch-on: version
+          cases:
+            1: u8
+            _: u4
+      - id: modification_time
+        type:
+          switch-on: version
+          cases:
+            1: u8
+            _: u4
+      - id: timescale
+        type: u4
+      - id: duration
+        type:
+          switch-on: version
+          cases:
+            1: u8
+            _: u4
+      - id: lang_code #TODO: decode (or describe how to decode)
+        type: u2
+      - id: predefined
+        type: u2
+  smhd_body:
+    seq:
+      - id: version
+        type: u4
+      - id: balance
+        type: s2
+      - id: reserved
+        type: u2
+  stco_body:
+    seq:
+      - id: version
+        type: u4
+      - id: entry_count
+        type: u4
+      - id: chunk_offsets
+        type: u4
+        repeat: expr
+        repeat-expr: entry_count
+  stsc_body:
+    seq:
+      - id: version
+        type: u4
+      - id: entry_count
+        type: u4
+      - id: entries
+        type: entry
+        repeat: expr
+        repeat-expr: entry_count
+    types:
+      entry:
+        seq:
+          - id: first_chunk
+            type: u4
+          - id: samples_per_chunk
+            type: u4
+          - id: sample_description_index
+            type: u4
+  stsd_body:
+    seq:
+      - id: version
+        type: u4
+      - id: entry_count
+        type: u4
+      - id: entries
+        type: entry
+        repeat: expr
+        repeat-expr: entry_count
+    types:
+      entry:
+        seq:
+          - id: size
+            type: u4
+          - id: type
+            type: str
+            size: 4
+          - id: body
+            type: entry_body
+            size: 'size - 8'
```

Comment:
`sizeof<u4> + 4` (when https://github.com/kaitai-io/kaitai_struct/issues/84 is implemented: `sizeof(size, type)`)